### ISSUE TITLE
Setup weblate for translation

### ISSUE THREAD

**Issue by ghost** [celzero/rethink-app#367]

**Comment by mezysinc** [comment]
more info:
https://docs.weblate.org/en/latest/faq.html

**Comment by ignoramous** [comment]
Thanks. @Ch4t4r did set it up for us... But the app first need to be purged of unused strings.

**Comment by Ch4t4r** [comment]
I could also help set up your own if you ever want to (otherwise I'm totally fine hosting it, but maybe you want something more "official" 😄)

**Comment by shuvashish76** [comment]
I agree Weblate is best as it's [opensource](https://github.com/WeblateOrg/weblate).

Features: https://weblate.org/en/features/
![Screenshot](https://user-images.githubusercontent.com/48226312/148335350-53094569-90c7-4752-a36e-e5f0470aaced.jpeg)
https://weblate.org/en/hosting/

**Comment by ignoramous** [comment]
https://hosted.weblate.org/engage/rethink-dns-firewall

**Comment by comradekingu** [comment]
@ignoramous Please don't use CC0. It is not a libre software license, and it causes all sorts of troubles for no benefit.
It is only very rarely public domain, which is not a good fit for a collaborative work with contributors from all sorts of jurisdictions.
Moreover, those are sometimes incompatible, and it is largely untested in all the relevant courts.

ISC is a very unproblematic license for what it is that is compatible with Apache 2.0. :)

**Comment by ignoramous** [comment]
> ISC is a very unproblematic license

ISC is license for code, not "documents" which is what these translation strings are. I have no qualms switching to MIT / ISC / CC-BY, but I had always thought CC-0 was a *fair*, no-holds-barred license.

> It is not a libre software license

Translations aren't code?

> It is only very rarely public domain

This is the first time I've heard CC-0 isn't similar to _public domain_. Can you please clarify further?

> ...which is not a good fit for a collaborative work with contributors from all sorts of jurisdictions

I know that in *certain* (limited) jurisdictions "public domain" doesn't exist, but that's not the norm in *most* places?

**Comment by comradekingu** [comment]
For something to actually be public domain, the person creating the work must be able to put it in the public domain.
That is not the case in a lot of jurisdictions.

>The laws of most jurisdictions throughout the world automatically confer exclusive Copyright and Related Rights (defined below) upon the creator and subsequent owner(s) (each and all, an "owner") of an original work of authorship and/or a database (each, a "Work").

Certain owners wish to permanently relinquish those rights to a Work for the purpose of contributing to a commons of creative, cultural and scientific works ("Commons") that the public can reliably and without fear of later claims of infringement build upon, modify, incorporate in other works, reuse and redistribute as freely as possible in any form whatsoever and for any purposes, including without limitation commercial purposes. These owners may contribute to the Commons to promote **the ideal** of a free culture and the further production of creative, cultural and scientific works, or to gain reputation or greater distribution for their Work in part through the use and efforts of others.

For these and/or other purposes and motivations, and without any expectation of additional consideration or compensation, the person associating CC0 with a Work (the "Affirmer"), to the extent that he or she is an owner of Copyright and Related Rights in the Work, voluntarily elects to apply CC0 to the Work and publicly distribute the Work under its terms, **with knowledge of his or her Copyright and Related Rights in the Work and the meaning and **intended legal effect** of CC0 on those rights**.

…

>2. Waiver. **To the greatest extent permitted by, but not in contravention of, applicable law**

So not only is it a social intent, but it hinges on legality. The idea is that contributors **should** know this, and the legal ramifications of their own legal system as they share it with others in the same jurisdiction, but also how it works when combined with the work of people from outside it.

The amount of people that know this are zero. We stop there and see what there is to gain, and there is no upside.
Known problems, lots of unknown ones, no gain to be had.

I could go on and on about how much I don't like the Creative Commons org, but the CC-BY 4.0 intl is OK.

ISC isn't a software license other than using the word "software". It is a good license for what it is trying to do.
Simple and readable permissive license with no strings attached.

I don't think MIT is good, but that is again more of me ranting.

**Comment by ignoramous** [comment]
Thanks.

ISC requires attribution, as does CC-BY-4.

Looking at something more closer to CC-0, 0-BSD, MIT-0 fit the bill. What do you think?

**Comment by comradekingu** [comment]
BSD-0 is what I thought ISC was, but it is somehow foregoes the `Copyright <YEAR> <OWNER> ` line, which I thought was what your problem was.
That part is a basic requirement of copyright (to cover jurisdictions where it isn't granted by default without doing anything), and
licensing is based on copyright.

<details>
Wasn't going to argue with you, but then even 2-Clause BSD has the same thing.

MIT-0 has it too, and it has the thing about MIT that I don't like removed, but then adds a CoC for its source repo, which I find counterproductive.
Yet another thing called MIT, which isn't well-written either.

0-BSD doesn't have it, so I guess if lawyers agree then problem solved.
Curiously 0-BSD is otherwise just ISC without "provided that the above copyright notice and this permission notice appear in all copies." I honestly don't understand how I could miss that.
Guess I bought the idea that ISC was the simplest it could be.

Aesthetically, it manages to be a longer license with
>Note: Despite its name, Zero-Clause BSD is an alteration of the ISC license, and is not textually derived from licenses in the BSD family. Zero-Clause BSD was originally approved under the name "Free Public License 1.0.0".
</details>
It is a confused name, with an alternate form, but that is as good as it gets.

TL;DR Ideally Apache 2.0 like the repo itself, or 0-BSD works.

**Comment by comradekingu** [comment]
Cleaning up the source strings a bit helps reduce the amount of confusion and extra translations/fixing to do.
I had a go in #619 which is a best-effort approach, but even so there are lots of wins to be had.

**Comment by ignoramous** [comment]
I'll take a look at #619 the first thing tomorrow, thanks.

I read some more about CC0, and it looks like it accounts for jurisdictions where public domain isn't recognised: https://opensource.stackexchange.com/questions/11221/can-the-0-clause-bsd-license-be-used-for-public-domain-work

Aside: Interesting discussion on MIT and xBSDs: https://opensource.stackexchange.com/questions/217/what-are-the-essential-differences-between-the-bsd-and-mit-licences
